jdaviz/core/marks.py

```diff
@@ -112,7 +113,11 @@ def set_y_unit(self, unit=None):
             if self.viewer.default_class is Spectrum1D:
                 spec = self.viewer.state.reference_data.get_object(cls=Spectrum1D)
                 eqv = u.spectral_density(spec.spectral_axis)
-                y = (self.y * self.yunit).to_value(unit, equivalencies=eqv)
+                if ('_pixel_scale_factor' in spec.meta):
+                    eqv += _eqv_pixar_sr(spec.meta['_pixel_scale_factor'])
+                    y = (self.y * self.yunit).to_value(unit, equivalencies=eqv)
+                else:
+                    y = (self.y * self.yunit).to_value(unit, equivalencies=eqv)
```

Comment:
`else` is unnecessary.
```suggestion
                if ('_pixel_scale_factor' in spec.meta):
                    eqv += _eqv_pixar_sr(spec.meta['_pixel_scale_factor'])
                y = (self.y * self.yunit).to_value(unit, equivalencies=eqv)
```